File: app/src/main/java/com/hanstcs/movieapp/movies/MoviesContract.java
package com.hanstcs.movieapp.movies;

import com.hanstcs.movieapp.base.BaseView;
import com.hanstcs.movieapp.data.Movies;

import java.util.List;

public interface MoviesContract {
    interface View extends BaseView<Presenter> {
        void showMovies(List<Movies> movies);
    }

    interface Presenter {
        void loadMovies();
    }
}

File: app/src/main/java/com/hanstcs/movieapp/movies/MoviesContract.kt
package com.hanstcs.movieapp.movies;

import com.hanstcs.movieapp.base.BaseView;
import com.hanstcs.movieapp.data.Movies;

interface MoviesContract {
    interface View : BaseView<Presenter> {
        fun showMovies(movies: MutableList<Movies>)
    }

    interface Presenter {
        fun loadMovies()
    }
}
